File: backend_dotnet/fruit_api/fruit_api/Models/Product.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fruit_api.Models;

[Table("Products")]
public class Product
{
    [Key]
    [Column("productId")]
    [StringLength(20)]
    public string ProductId { get; set; } = null!;

    [Required]
    [Column("categoryId")]
    public string CategoryId { get; set; } = null!;

    [Required]
    [Column("supplierId")]
    public string SupplierId { get; set; } = null!;

    [Required]
    [MaxLength(150)]
    [Column("productName")]
    public string ProductName { get; set; } = string.Empty;

    [Required]
    [MaxLength(50)]
    [Column("unit")]
    public string Unit { get; set; } = string.Empty;

    [Required]
    [Column("price", TypeName = "decimal(12,2)")]
    public decimal Price { get; set; }

    [Column("stockQuantity")]
    public int StockQuantity { get; set; }

    [Column("description")]
    public string? Description { get; set; }

    [MaxLength(255)]
    [Column("imageUrl")]
    public string? ImageUrl { get; set; }

    [Column("isActive")]
    public bool IsActive { get; set; } = true;

    [Column("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    // Navigation properties
    public Category? Category { get; set; }
    public Supplier? Supplier { get; set; }
    public ICollection<CartItem>? CartItems { get; set; }
    public ICollection<OrderItem>? OrderItems { get; set; }
    public ICollection<Review>? Reviews { get; set; }
    public ICollection<Favorite>? Favorites { get; set; }
}

File: frontend_flutter/web_admin/lib/models/product.dart
class Product {
  final String productId;
  final String productName;
  final String? categoryId;
  final String? categoryName;
  final String? supplierId;
  final String? supplierName;
  final String unit;
  final double price;
  final int stockQuantity;
  final String? description;
  final String? imageUrl;
  final bool isActive;
  final DateTime createdAt;
  final double averageRating;
  final int reviewCount;

  Product({
    required this.productId,
    required this.productName,
    this.categoryId,
    this.categoryName,
    this.supplierId,
    this.supplierName,
    required this.unit,
    required this.price,
    required this.stockQuantity,
    this.description,
    this.imageUrl,
    required this.isActive,
    required this.createdAt,
    this.averageRating = 0,
    this.reviewCount = 0,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      productId: json['productId'] ?? '',
      productName: json['productName'] ?? '',
      categoryId: json['categoryId'],
      categoryName: json['categoryName'],
      supplierId: json['supplierId'],
      supplierName: json['supplierName'],
      unit: json['unit'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      stockQuantity: json['stockQuantity'] ?? 0,
      description: json['description'],
      imageUrl: json['imageUrl'],
      isActive: json['isActive'] ?? true,
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
      averageRating: (json['averageRating'] ?? 0).toDouble(),
      reviewCount: json['reviewCount'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'productId': productId,
      'productName': productName,
      'categoryId': categoryId,
      'categoryName': categoryName,
      'supplierId': supplierId,
      'supplierName': supplierName,
      'unit': unit,
      'price': price,
      'stockQuantity': stockQuantity,
      'description': description,
      'imageUrl': imageUrl,
      'isActive': isActive,
      'createdAt': createdAt.toIso8601String(),
      'averageRating': averageRating,
      'reviewCount': reviewCount,
    };
  }
}